lib/EasyPost/Service/CarrierAccountService.php

```diff
@@ -43,7 +43,16 @@ public function all(mixed $params = null): mixed
      */
     public function update(string $id, mixed $params): mixed
     {
-        return self::updateResource(self::serviceModelClassName(self::class), $id, $params);
+        $carrierAccount = self::retrieve($id);
+        $carrierAccountType = $carrierAccount['type'];
+        if (in_array($carrierAccountType, Constants::UPS_ACCOUNT_TYPES_WITH_CUSTOM_WORKFLOWS, true)) {
+            $updateUrl = 'upsOauthRegistration';
+            $params = [self::selectTopLayerKey($carrierAccountType) => $params];
+        } else {
+            $updateUrl = 'carrierAccount';
```

Comment:
These are not URLs and should not be named as such. They become a URL later but are actually class names here. Also, wouldn't they be title cased?

```suggestion
            $className = 'UpsOauthRegistration';
            $params = [self::selectTopLayerKey($carrierAccountType) => $params];
        } else {
            $className = 'CarrierAccount';
```

Reply:
I don't think it's case sensitive, but I have uppercased the first character

Reply:
Class names must be capitalized, otherwise it's not a true class name.